crates/runtime/src/dataconnector.rs

```diff
@@ -91,6 +92,9 @@ pub enum Error {
     UnableToFilterDataFrame {
         source: datafusion::error::DataFusionError,
     },
+
+    #[snafu(display("{message}"))]
+    UnableToCreateDataConnector { message: String },
```

Comment:
If the only usage of this error is to report that we couldn't connect - I think it would be better to bring the error message into this, rename it to something specific and only pass the two properties needed to display in the error (host/port)

Reply:
Yes, but this is a generic error for all data connectors, and in general case it can have only common prefix, but reason will be different in each case